src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.PreferConstCharOverConstUnitStringInStringBuilderMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.PreferConstCharOverConstUnitStringInStringBuilderDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Performance,
+                                                                                      RuleLevel.IdeSuggestion,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            InvocationExpressionSyntax invocationExpression = (InvocationExpressionSyntax)context.Node;
+
+            if (!(invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression))
+            {
+                return;
+            }
+
+            // Verify that the current call is to StringBuilder.Append
+            if (memberAccessExpression.Name.ToString() != nameof(StringBuilder.Append))
+            {
+                return;
+            }
+
+            // Check that we've called Append on an instance of StringBuilder
+            ISymbol memberSymbol = context.SemanticModel.GetSymbolInfo(memberAccessExpression).Symbol;
+            if (memberSymbol == null)
+            {
+                return;
+            }
+
+            string memberSymbolContainingTypeString = memberSymbol.ContainingType.ToString();
```

Comment:
1. Unused. 
2. You should not check types using the name, instead you should try to retrieve an instance of the type symbol and check for equality. See https://github.com/dotnet/roslyn-analyzers/pull/3387/files#diff-009312362ad767f47f1855e7de22a6c7R38-R43 and https://github.com/dotnet/roslyn-analyzers/pull/3387/files#diff-009312362ad767f47f1855e7de22a6c7R55

Reply:
Fixed.